# Question 10
-----

In [1]:
mean = 2122 / 80
var = (80/79) * ((56384 / 80) - (mean**2))
print("a)")
print("mean = " + str(mean))
print("var = " + str(var))

a)
mean = 26.525
var = 1.2398734177215283


In [2]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [3]:
z = normal(0, 1)
loc_1 = z.locate(0.005)
loc_2 = -1 * loc_1
sigma = math.sqrt(var) / math.sqrt(80)
LI = (loc_1 * sigma) + mean
UI = (loc_2 * sigma) + mean
print("b) " + str(LI) + " < X < " + str(UI))

b) 26.204330609773734 < X < 26.845669390226263


c) The manufacturer is justified, as the breaking strength of the yarn is increased to values between 26.2 and 26.8

d) It was assumed that the distribution was normal